### ISSUE TITLE
False positive in adhoc script

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/SqlServer.Rules#731]
<img width="630" height="39" alt="Image" src="https://github.com/user-attachments/assets/30b68fed-1dd9-4d9f-a62f-ea2b82f8c6fe" />

```dos
Severity	Code	Description	Project	File	Line	Suppression State
Warning (active)	SRD0063	Do not use IF statements containing queries in stored procedures.		C:\dev\Database.Internal\DataLoss\DataLossScript.sql	30
```

```sql
IF OBJECT_ID('dbo.PurchaseOrderLineSplit', 'U') IS NOT NULL
BEGIN
    IF COL_LENGTH('dbo.PurchaseOrderLineSplit', 'Has2DCode') IS NOT NULL
    BEGIN
        ALTER TABLE dbo.PurchaseOrderLine ADD [Has2DCode] BIT NULL;
        UPDATE dbo.PurchaseOrderLine SET [Has2DCode] = 1;
        ALTER TABLE dbo.PurchaseOrderLineSplit DROP COLUMN [Has2DCode];
    END
    IF COL_LENGTH('dbo.PurchaseOrderLineSplit', 'ActualQuantity') IS NOT NULL
    BEGIN
        ALTER TABLE dbo.PurchaseOrderLineSplit DROP COLUMN [ActualQuantity];
    END
END
```

**Comment by ErikEJ** [comment]
This is probably due to syntetic stored procedure handling
